Compiler/script2/cs_parser.cpp

```diff
@@ -1620,28 +1618,26 @@ void AGS::Parser::ParseFuncdecl(size_t declaration_start, TypeQualifierSet tqs,
     }
 
     // Stash away the known info about the function so that we can check whether this declaration is compatible
-    SymbolTableEntry::FunctionDesc *known_info = _sym[name_of_func].FunctionD;
+    std::shared_ptr<SymbolTableEntry::FunctionDesc> known_info{ _sym[name_of_func].FunctionD };
```

Comment:
The commit's description mentions unique_ptr, which would be correct, but you are using shared_ptr here, which is redundant as the object is not shared and disposed soon after.